packages/sessions-sdk-react/src/token-list.tsx

```diff
@@ -0,0 +1,164 @@
+import { PaperPlaneTiltIcon } from "@phosphor-icons/react/dist/ssr/PaperPlaneTilt";
+import { WalletIcon } from "@phosphor-icons/react/dist/ssr/Wallet";
+import { motion } from "motion/react";
+import { GridList, GridListItem } from "react-aria-components";
+
+import { amountToString } from "./amount-to-string.js";
+import { Button } from "./button.js";
+import { FetchError } from "./fetch-error.js";
+import { Link } from "./link.js";
+import type { EstablishedSessionState } from "./session-provider.js";
+import styles from "./token-list.module.css";
+import { useFaucet } from "./use-faucet.js";
+import type { Token } from "./use-token-account-data.js";
+import {
+  StateType as TokenDataStateType,
+  useTokenAccountData,
+} from "./use-token-account-data.js";
+
+const MotionGridList = motion(GridList<Token>);
+
+type Props = {
+  sessionState: EstablishedSessionState;
+  onPressReceiveTokens: () => void;
+} & (
+  | { onPressToken: (token: Token) => void }
+  | { onPressSend: (token: Token) => void }
+);
+
+export const TokenList = ({
+  sessionState,
+  onPressReceiveTokens,
+  ...props
+}: Props) => {
+  const { faucetUrl, showFaucet } = useFaucet(sessionState);
+  const state = useTokenAccountData(sessionState);
+  switch (state.type) {
+    case TokenDataStateType.Error: {
+      return (
+        <FetchError
+          headline="Failed to fetch your wallet balances"
+          error={state.error}
+          reset={state.reset}
+          className={styles.tokenListError}
+        />
+      );
+    }
+    case TokenDataStateType.Loaded: {
+      return state.data.tokensInWallet.length === 0 ? (
+        <div className={styles.tokenListEmpty}>
+          <WalletIcon className={styles.emptyIcon} />
+          <span className={styles.message}>Your wallet is empty</span>
+          <span className={styles.hints}>
+            <Link onPress={onPressReceiveTokens}>Receive</Link> or{" "}
+            <Link
+              onPress={showFaucet}
```

Comment:
should this generalize to mainnet? like direct users to bridge instead of faucet?